src/main/java/org/jboss/snowdrop/licenses/LicensesGenerator.java

```diff
@@ -120,4 +118,15 @@ private Artifact gavToArtifact(Gav gav) {
                 gav.getType());
     }
 
+
+    protected static LicenseSummaryFactory createLicenseSummaryFactory(GeneratorProperties generatorProperties) {
```

Comment:
Should be private

Reply:
It would be nice to have it private. 
I made it protected to use id here: https://github.com/snowdrop/licenses-generator/pull/1/files#diff-694aa41022790d3e6022b5a14957ac1fR59
I don't want to copy this code. What would be the best way to create it for the test if it were private?

Reply:
You could implement that test using mocks similar to https://github.com/snowdrop/licenses-generator/blob/master/src/test/java/org/jboss/snowdrop/licenses/sanitiser/ExceptionLicenseSanitiserTest.java and https://github.com/snowdrop/licenses-generator/blob/master/src/test/java/org/jboss/snowdrop/licenses/sanitiser/AliasLicenseSanitiserTest.java. This way your test would explicitly test that one sanitiser (this should be the point) and woulnd't depend on any other class.

Reply:
I wanted to test the whole flow to be sure the proper license is selected for each case (when something is in the service, when it is in the exception file, etc).

Reply:
For the full flow we should create integration tests which would check files too.